aries_cloudagent/ledger/base.py

```diff
@@ -75,6 +75,23 @@ async def get_all_endpoints_for_did(self, did: str) -> dict:
             did: The DID to look up on the ledger or in the cache
         """
 
+    @abstractmethod
+    async def construct_attr_json(
```

Comment:
The implementation in the concretions for this method are exactly the same (right now, at least -- perhaps there's a reason for there to be differences in the future). It might make sense to move the implementation of this method to the base class here and just have it not be an abstract method. Also, just by convention, since we don't really intend this method to be called outside of the ledger class, it might be more appropriate to consider it "private" and name it `_construct_attr_json`.